### ISSUE TITLE
Machine Learning material

### ISSUE THREAD

**Issue by jeannesorin** [OpenSourceEcon/BootCamp2019#14]
What material do you have on machine learning?

**Comment by Xincheng-Qiu** [comment]
The 2018 AEA Continuing Education Webcasts on Machine Learning and Econometrics by Susan Athey and Guido Imbens might be a place to start~
https://www.aeaweb.org/conference/cont-ed/2018-webcasts

**Comment by rickecon** [comment]
Great reference above to the Susan Athey material. I have taught intro to machine learning for economists every year in the M.A. Program in Computational Economics at the University of Chicago. My most recent Jupyter notebooks are the [`persp-model-econ_W19/Notebooks/`](https://github.com/UC-MACSS/persp-model-econ_W19/tree/master/Notebooks) directory of my [modeling course repository](https://github.com/UC-MACSS/persp-model-econ_W19). You might just want to fork and clone the whole repo. There are some nice problem sets in this repo. I have listed the machine learning notebooks below with a short description. But I will be updating these notebooks and covering them in the last week of the boot camp. In particular, I will reference some of the content covered by John Rust, Sanjog Misra, and Whitney Newey in the DSE Summer School today and yesterday.

1. [Classification](https://github.com/UC-MACSS/persp-model-econ_W19/blob/master/Notebooks/Classification/LogitKNN.ipynb) notebook. This notebook goes through logit, multinomial logit, and K Nearest Neighbors.
2. [Resampling methods](https://github.com/UC-MACSS/persp-model-econ_W19/blob/master/Notebooks/Resampling/Resampling.ipynb). How to implement the paradigm of cross validation through k-folds and bootstrapping of estimating your model multiple times on training sets and measuring accuracy on test set. This is one of the most important contributions of the machine learning advances in economics, almost as important as the new models.
3. [Tree-based methods](https://github.com/UC-MACSS/persp-model-econ_W19/blob/master/Notebooks/Trees/Trees.ipynb). This notebook goes through decision trees and random forest methods. These end up being powerful machine learning methods that sometimes beat neural nets in accuracy. Also, this notebook introduces the concept of tuning hyperparameters to maximize test set accuracy. This is a powerful concept that is the basis for what TensorFlow is good at.
4. [Support vector machines](https://github.com/UC-MACSS/persp-model-econ_W19/blob/master/Notebooks/SVM/SVM.ipynb). SVM is an important classifier.
5. [Neural nets](https://github.com/UC-MACSS/persp-model-econ_W19/blob/master/Notebooks/NeuralNet/NeuralNet.ipynb). This is an introduction to neural networks. The notebook takes you through the multi-layer perceptron (MLP) model that includes hidden layer (deep nets).
